syntax = "proto3";

package downloader;

service DownloaderService {
  rpc GetVideoMetadata(DownloadRequest) returns (VideoMetadataResponse);
  rpc DownloadVideo(DownloadRequest) returns (stream DownloadStatusResponse);
}

message DownloadRequest {
  string video_url = 1;
}

message DownloadStatusResponse {
  oneof status {
    float progress_percentage = 1; // Para o progresso de 0 a 100
    string final_message = 2;      // Para a mensagem final de sucesso
    string error_message = 3;      // Para a mensagem final de erro
  }
}

message VideoMetadataResponse {
  string title = 1;
  int32 duration = 2;       
  string thumbnail_url = 3;
}